QUARKUS-2994.md

```diff
@@ -0,0 +1,43 @@
+# QUARKUS-2994 JDK 21 Support
+
+This document is a high level test plan for certification of Red Hat build of Quarkus on JDK 21.
+
+## Scope of the testing
+
+### Current test coverage
+* Regular CI runs in upstream include full testing for Java 21 (according to the announcement)
+* Daily runs on java 21 in GitHub Actions
+* Checks of code.quarkus in StartStop TS
+
+### Impact on test suites and testing automation
+* Temurin JVM 21 and OpenJDK 21 need to be added as JDKs to Quarkus QE Jenkins
+* Temurin JVM 21 and OpenJDK 21 have to be added as JVMs to Polarion
+* Bare metal jobs in extended platform testing need to be extended to run with both Temurin 21 and OpenJDK 21.
+  * This affects JVM mode only.
+* Openshift JVm job needs to use OpenJDK 21 (together with existing 17) instead of OpenJDK 11
+  * We also should use s2i build image based on jdk 21 there
+* Two quickstart jobs (Linux and Windows) should be added/repurposed from 11 in general or extended pipeline
+* Add a job for running on ARM
+
+### Impact on resources
+* Should not change (or be minimal), since for affected jobs we should drop testing on java 11, so consumed resource will stay the same.
+* In general support for JDK depends on number of used jobs. For basic product support, each run requires the following resources on Openstack:
+  *  Baremetal 3 combinations (podman, docker, fips) * 8 scenarios * `large`(8192 RAM + 4 vCPU) node
+  *  openshift two combinations (eus, stable) * `medium`(4096 RAM + 4 vCPU) node.
+    * Each combination includes ~180 tests (running together for 40 minutes), each creates and deletes a dedicated Openshift namespace of unknown resource consumption  
+  * Quickstarts on RHEL: `large` jenkins node (see above)
+  * Quickstarts on Windows: `medium` jenkins node and `m5zn.metal` EC2 instance
+  * ARM job: 8 scenarios *  large node + OCP cluster on Amazon
+
```

Comment:
Please add `Impact on documentation`.

There is https://issues.redhat.com/browse/QUARKUS-3970 linked under JDK 21 Epic.

You need to check what guides are referring to JDK versions and make sure that the guides for 3.8 refer to JDK 17/21, and mentions of JDK 11 are not present. RN should mention drop of JDK 11.